zerver/tests/test_realm_filters.py

```diff
@@ -117,3 +117,71 @@ def test_delete(self) -> None:
+            realm, filter_id, "#(?P<id>[0-9]+)", "https://realm.com/my_realm_filter/issues/%(id)s"
+        )
+        result = self.client_get("/json/realm/filters")
+        self.assert_json_success(result)
+        self.assertEqual(200, result.status_code)
+        self.assertEqual(len(result.json()["filters"]), 1)
+
+        data = {"pattern": "", "url_format_string": "https://realm.com/my_realm_filter/%(id)s"}
+        result = self.client_patch(f"/json/realm/filters/{filter_id}", info=data)
+        self.assert_json_error(result, "This field cannot be blank.")
+
+        data["pattern"] = "$a"
+        result = self.client_patch(f"/json/realm/filters/{filter_id}", info=data)
+        self.assert_json_error(
+            result, "Invalid filter pattern.  Valid characters are [ a-zA-Z_#=/:+!-]."
+        )
+
+        data["pattern"] = r"ZUL-(?P<id>\d++)"
+        result = self.client_patch(f"/json/realm/filters/{filter_id}", info=data)
+        self.assert_json_error(
+            result, "Invalid filter pattern.  Valid characters are [ a-zA-Z_#=/:+!-]."
+        )
+
+        data["pattern"] = r"ZUL-(?P<id>\d+)"
+        data["url_format_string"] = "$fgfg"
+        result = self.client_patch(f"/json/realm/filters/{filter_id}", info=data)
+        self.assert_json_error(result, "Enter a valid URL.")
+
+        data["pattern"] = r"ZUL-(?P<id>\d+)"
+        data["url_format_string"] = "https://realm.com/my_realm_filter/"
+        result = self.client_patch(f"/json/realm/filters/{filter_id}", info=data)
+        self.assert_json_error(result, "Invalid URL format string.")
+
+        data["url_format_string"] = "https://realm.com/my_realm_filter/#hashtag/%(id)s"
+        result = self.client_patch(f"/json/realm/filters/{filter_id}", info=data)
+        self.assert_json_success(result)
+        self.assertIsNotNone(re.match(data["pattern"], "ZUL-15"))
+
+        data["pattern"] = r"_code=(?P<id>[0-9a-zA-Z]+)"
+        data["url_format_string"] = "https://example.com/product/%(id)s/details"
+        result = self.client_patch(f"/json/realm/filters/{filter_id}", info=data)
+        self.assert_json_success(result)
+        self.assertIsNotNone(re.match(data["pattern"], "_code=123abcdZ"))
+
+        data["pattern"] = r"PR (?P<id>[0-9]+)"
+        data[
+            "url_format_string"
+        ] = "https://example.com/~user/web#view_type=type&model=model&action=12345&id=%(id)s"
+        result = self.client_patch(f"/json/realm/filters/{filter_id}", info=data)
+        self.assert_json_success(result)
+        self.assertIsNotNone(re.match(data["pattern"], "PR 123"))
+
+        data["pattern"] = r"(?P<org>[a-zA-Z0-9_-]+)/(?P<repo>[a-zA-Z0-9_-]+)#(?P<id>[0-9]+)"
+        data["url_format_string"] = "https://github.com/%(org)s/%(repo)s/issue/%(id)s"
+        result = self.client_patch(f"/json/realm/filters/{filter_id}", info=data)
+        self.assert_json_success(result)
+        self.assertIsNotNone(re.match(data["pattern"], "zulip/zulip#123"))
```

Comment:
Are these tests basically a copy of the set of values from the "Create a linkifier" test?  Probably it's not worth duplicating them all, given that we know structurally we're running the exact same validation code in this code path.

Reply:
I have reduced the number of test cases and have just put the basic test cases now.